src/video/SDL_surface.c

```diff
@@ -60,14 +76,22 @@ SDL_Surface *
 SDL_CreateRGBSurfaceWithFormat(Uint32 flags, int width, int height, int depth,
                                Uint32 format)
 {
-    Sint64 pitch;
+    size_t pitch;
     SDL_Surface *surface;
 
     /* The flags are no longer used, make the compiler happy */
     (void)flags;
 
+    if (width < 0) {
+        return SDL_InvalidParamError("width");
```

Comment:
```
./src/video/SDL_surface.c(86): Warning! W102: Type mismatch (warning)
./src/video/SDL_surface.c(86): Note! I2003: source conversion type is 'int '
./src/video/SDL_surface.c(86): Note! I2004: target conversion type is 'struct SDL_Surface *'
./src/video/SDL_surface.c(90): Warning! W102: Type mismatch (warning)
./src/video/SDL_surface.c(90): Note! I2003: source conversion type is 'int '
./src/video/SDL_surface.c(90): Note! I2004: target conversion type is 'struct SDL_Surface *'
./src/video/SDL_surface.c(209): Warning! W102: Type mismatch (warning)
./src/video/SDL_surface.c(209): Note! I2003: source conversion type is 'int '
./src/video/SDL_surface.c(209): Note! I2004: target conversion type is 'struct SDL_Surface *'
./src/video/SDL_surface.c(213): Warning! W102: Type mismatch (warning)
./src/video/SDL_surface.c(213): Note! I2003: source conversion type is 'int '
./src/video/SDL_surface.c(213): Note! I2004: target conversion type is 'struct SDL_Surface *'
./src/video/SDL_surface.c(218): Warning! W102: Type mismatch (warning)
./src/video/SDL_surface.c(218): Note! I2003: source conversion type is 'int '
./src/video/SDL_surface.c(218): Note! I2004: target conversion type is 'struct SDL_Surface *'
./src/video/SDL_surface.c(246): Warning! W102: Type mismatch (warning)
./src/video/SDL_surface.c(246): Note! I2003: source conversion type is 'int '
./src/video/SDL_surface.c(246): Note! I2004: target conversion type is 'struct SDL_Surface *'
./src/video/SDL_surface.c(250): Warning! W102: Type mismatch (warning)
./src/video/SDL_surface.c(250): Note! I2003: source conversion type is 'int '
./src/video/SDL_surface.c(250): Note! I2004: target conversion type is 'struct SDL_Surface *'
./src/video/SDL_surface.c(255): Warning! W102: Type mismatch (warning)
./src/video/SDL_surface.c(255): Note! I2003: source conversion type is 'int '
./src/video/SDL_surface.c(255): Note! I2004: target conversion type is 'struct SDL_Surface *'
```
You are supposed to return `NULL` here, not an error integer.

Reply:
Oops, yes. Now fixed.